public/js/application.js

```diff
@@ -0,0 +1,114 @@
+$(document).ready(function() {
+  $('#guess').click(function(){
+    guess = prompt("Guess the secret code to reveal a json object in the javascript console.");
+    $.ajax({
+      url: "/.json",
+      type: "get",
+      data: {code: guess},
+      success: function(callback){
```

Comment:
Please dont call this callback - callback implies its a function.  Call this response.